File: Data-Structures-And-Libraries/Python/URI-1258.py
def reset_state(dicitonary) -> None:
    for element in dicitonary.keys():
        dicitonary[element] = []


def sort_dict(dictionary) -> None:
    for index in dictionary.keys():
        dictionary[index] = sorted(dictionary[index])


def main():

    """ dictionaries for solving the problem """

    color_size_dict = {
        "brancoP": 1,
        "brancoM": 2,
        "brancoG": 3,
        "vermelhoP": 4,
        "vermelhoM": 5,
        "vermelhoG": 6,
    }

    owners_dict = {
        1: [],
        2: [],
        3: [],
        4: [],
        5: [],
        6: [],
    }

    relate_dict = {
        1: "branco P",
        2: "branco M",
        3: "branco G",
        4: "vermelho P",
        5: "vermelho M",
        6: "vermelho G",
    }

    to_jump = 0

    while True:

        cases = int(input())
        if cases == 0:
            break
        if to_jump != 0:
            print()
        for i in range(cases):
            name = input()
            attributes = input().split()

            case_number = color_size_dict[attributes[0] + attributes[1]]
            owners_dict[case_number].append(name)

        sort_dict(owners_dict)

        for number, lst in owners_dict.items():
            if len(lst) == 0:
                continue
            for owner in lst:
                print(relate_dict[number], owner)

        reset_state(owners_dict)
        to_jump += 1

main()

File: Data-Structures-And-Libraries/Java/URI-1258.java
import java.io.IOException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;
 
public class Main {
 
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
 
		/*Maps for solving the problems */
 
		Map<Integer, ArrayList<String>> owners = new HashMap<Integer, ArrayList<String>>(){{
			put(1, new ArrayList<String>());
			put(2, new ArrayList<String>());
			put(3, new ArrayList<String>());
			put(4, new ArrayList<String>());
			put(5, new ArrayList<String>());
			put(6, new ArrayList<String>());		
		}};
 
		Map<String, Integer> colorSize = new HashMap<String, Integer>(){{
			put("brancoP", 1);
			put("brancoM", 2);
			put("brancoG", 3);
			put("vermelhoP", 4);
			put("vermelhoM", 5);
			put("vermelhoG", 6);			
		}};
 
		Map<Integer, String> related = new HashMap<Integer, String>(){{
			put(1, "branco P");
			put(2, "branco M");
			put(3, "branco G");
			put(4, "vermelho P");
			put(5, "vermelho M");
			put(6, "vermelho G");
		}};
 
		int cases = sc.nextInt();
		boolean lineControl = false;
 
		do {
			sc.nextLine();
			if(lineControl) System.out.println();
 
			for(int i = 0; i < cases; i++) {
				String name = sc.nextLine();
				String[] attributes = sc.nextLine().split(" ");
 
				int caseNumber = colorSize.get(attributes[0] + attributes[1]);
				owners.get(caseNumber).add(name);
			}
 
			sortDict(owners);
			/* all good till here */
 
			for(Map.Entry<Integer, ArrayList<String>> entry : owners.entrySet()) {
				if(entry.getValue().isEmpty()) {
					continue;
				}
				for(String owner : entry.getValue()) {
					System.out.printf("%s %s%n", related.get(entry.getKey()), owner);				
				}					
			}
 
			resetState(owners);
			lineControl = true;
 
			cases = sc.nextInt();
 
		}while(cases != 0);
 
	}
 
	static void resetState(Map<Integer, ArrayList<String>> ownersDict) {
		for(Integer element: ownersDict.keySet()) {
			ownersDict.put(element, new ArrayList<String>());
		}
 
	}
 
	static void sortDict(Map<Integer, ArrayList<String>> ownersDict) {
		for(Integer element : ownersDict.keySet()) {
			Collections.sort(ownersDict.get(element));		
		}
 
	}
}
